CONTRIBUTING.md

```diff
@@ -1,25 +1,114 @@
-to access data within the error. The return types for any source errors are `dyn Error`, which
-allows us to update dependencies without breaking the public API.
+### Issues
 
-Normally, the fields of the error struct are not public, as they can be accessed through the
-provided methods. However, in some cases, you may need to make the struct fields `pub` or
-`pub(crate)`.
+There are two major kinds of issues in Twilight, user-created and
+developer-created. Of the user-created issues, they are usually bug reports,
+feature requests, or support requests. If reporting a bug, please include the
+operating system used, any relevant information about the tech stack, and the
+feature flags used. Before making a support issue, be sure to consider joining
+the [Twilight Discord] and making a thread for your question in the `#support`
+channel.
 
-Any new error implementations must follow this pattern:
+Developer-created issues are most often bugs or tracking issues for supporting
+Discord API features. We aim for 100% coverage of the Discord API; we will wait
+until a new feature is supported in the [Discord API] before adding its support
+to Twilight.
+
+### Pull Requests
+
+In order to ease developer cooperation, please make a pull request from a
+feature branch on your own fork instead of from `main` or `next`. Additionally,
+since we squash every commit before merging, force-pushing a branch with new
+changes or updates is not always required. This helps with maintaining a
+chronological history in the PR story.
+
+Contributors must add tests and documentation that reflects their changes.
+
+#### Format
+
+Twilight follows a [Conventional Commit] style for pull request titles and
+commits, using [Angular]'s types.
+
+The title of the PR must follow this style:
+
+`<type>(<scope>)<! if breaking>: <short summary>`
+
+Acceptable values of the `type` field:
+- `build`: Changes that affect the build system or external dependencies (example
+  scopes: gulp, broccoli, npm)
+- `ci`: Changes to our CI configuration files and scripts (examples: CircleCi,
+  SauceLabs)
+- `docs`: Documentation only changes
+- `feat`: A new feature
+- `fix`: A bug fix
+- `perf`: A code change that improves performance
+- `refactor`: A code change that neither fixes a bug nor adds a feature
+- `test`: Adding missing tests or correcting existing tests
+
+Acceptable scopes are any crate in the project. Add an exclamation point after
+the scope (outside the parentheses) if the change is breaking.
+
+The PR description must contain a short description of the changes.
+
+#### Labeling
+
+Labels that must be added manually:
+- `d-api`: for changes that correspond with the [Discord API]
+- `d-breaking`: for API changes that cause a breaking change in library code
+- `d-unmerged`: changes that have not yet been merged into the documentation
+- `w-do-not-merge`: merge is blocked on another PR, or is not ready for review
```

Comment:
I am not sure if this is correct wording here, as I understand it, it should be used for PR's that should not be merged currently, but in other ways are ready. PR's that are not ready for review should be a draft.